dask_geopandas/backends.py

```diff
@@ -14,6 +13,16 @@
 import geopandas
 from geopandas.array import GeometryArray, GeometryDtype, from_shapely
 
+DASK_2021_06 = str(dask.__version__) > LooseVersion("2021.05.0")
+
+if DASK_2021_06:
+    from dask.dataframe.utils import make_meta_obj as make_meta
```

Comment:
With latest dask, `make_meta_obj` is a dispatch to handle meta creation for arbitrary object (for ex: `object`, `int`, `dict`..). `make_meta_obj` is being used to register against `geopandas.GeoSeries` & `geopandas.GeoDataFrame` - This will lead to incorrect behavior which we are seeing now in CI. Instead it has to be something like:

```python
from dask.dataframe.utils import make_meta, make_meta_obj
@make_meta.register((geopandas.GeoSeries, geopandas.GeoDataFrame))
def make_meta_geodataframe(df, index=None):
    return df.head(0)

@make_meta_obj.register((geopandas.GeoSeries, geopandas.GeoDataFrame)):
def make_meta_object(x, index=None):
    if is_arraylike(x) and x.shape:
        return x[:0]

    if index is not None:
        index = make_meta(index)

    if isinstance(x, dict):
        return pd.DataFrame(
            {c: _empty_series(c, d, index=index) for (c, d) in x.items()}, index=index
        )
    if isinstance(x, tuple) and len(x) == 2:
        return _empty_series(x[0], x[1], index=index)
    elif isinstance(x, (list, tuple)):
        if not all(isinstance(i, tuple) and len(i) == 2 for i in x):
            raise ValueError(
                "Expected iterable of tuples of (name, dtype), got {0}".format(x)
            )
        return pd.DataFrame(
            {c: _empty_series(c, d, index=index) for (c, d) in x},
            columns=[c for c, d in x],
            index=index,
        )
    elif not hasattr(x, "dtype") and x is not None:
        # could be a string, a dtype object, or a python type. Skip `None`,
        # because it is implictly converted to `dtype('f8')`, which we don't
        # want here.
        try:
            dtype = np.dtype(x)
            return _scalar_from_dtype(dtype)
        except Exception:
            # Continue on to next check
            pass

    if is_scalar(x):
        return _nonempty_scalar(x)

    raise TypeError("Don't know how to create metadata from {0}".format(x))
```